test-stats.py

```diff
@@ -0,0 +1,137 @@
+                    'missing': '',
+                    'pending': {'status': '', 'url': '', 'user': {}},
+                    'test_tags': {'classic': '', 'container': '', 'atomic': ''}},
+                'upstreamfirst': {'test_yml': '',
+                                  'package_url': '',
+                                  'test_tags': {'classic': '', 'container': '', 'atomic': ''}}}
+
+
+def wrapper_for_get_pull_requests(package):
+    """Function updates package_json with correct information"""
+
+    raw_text = get_pull_requests(DIST_GIT_URL, package)
+    result = manage_pull_request(raw_text)
+    package_json['name'] = package
+
+    if result is not None:
+        package_json['distgit']['pending']['url'] = result['url']
+        package_json['distgit']['pending']['user'] = result['user']
+        package_json['distgit']['pending']['status'] = True
+    else:
+        package_json['distgit']['pending']['url'] = ''
+        package_json['distgit']['pending']['user'] = ''
+        package_json['distgit']['pending']['status'] = False
+
+    # Get upstreamfirst test-tags
+    upstream_test_tags = handle_test_tags(UPSTREAMFIRST_URL, package)
+    upstream_test_tags = test_tags_to_dict(upstream_test_tags)
+    package_json['upstreamfirst']['test_tags'] = upstream_test_tags
+
+    upstream_url_to_test_yml = get_url_to_test_yml(UPSTREAMFIRST_URL, package)
+    if check_if_test_yml_exists(upstream_url_to_test_yml):
+        package_json['upstreamfirst']['test_yml'] = True
+        package_json['upstreamfirst']['package_url'] = upstream_url_to_test_yml
+    else:
+        package_json['upstreamfirst']['test_yml'] = False
+        package_json['upstreamfirst']['package_url'] = ''
+
+    # Get distgit test-tags
+    dist_git_test_tags = handle_test_tags(DIST_GIT_URL, package)
+    dist_git_test_tags = test_tags_to_dict(dist_git_test_tags)
+    package_json['distgit']['test_tags'] = dist_git_test_tags
+
+    dist_git_url_to_test_yml = get_url_to_test_yml(DIST_GIT_URL, package)
+    if check_if_test_yml_exists(dist_git_url_to_test_yml):
+        package_json['distgit']['test_yml'] = True
+        package_json['distgit']['package_url'] = dist_git_url_to_test_yml
+    else:
+        package_json['distgit']['test_yml'] = False
+        package_json['distgit']['package_url'] = ''
+
+    # Write results to the file
+    write_results_to_the_file(package_json, config.FILE_PACKAGES)
+
+
+def get_projects_list_from_url(site_url):
+    # Get all open pull requests
+    projects = get_all_projects(UPSTREAMFIRST_URL)
+    all_packages = get_projects_names(projects)
+    return all_packages
+
+
+def create_file_with_packages_data(input_data, short=False):
+    # Create file with packages_data input data is a file or an URL. Filename is config.FILE_PACKAGES
+    if 'https://' in input_data:
+        all_packages = get_projects_list_from_url(input_data)
+    else:
+        all_packages = get_list_of_packages_from_the_file(input_data)
+
+    if short:
```

Comment:
Purpose of short is time optimisation. What is a reason to gather all information about all packages and out only for 10 first. Current script version process only first 10 packages + skips parsing other packages.
In this case script will run only a few seconds.

Reply:
Actually, it's one API request:
'https://upstreamfirst.fedorainfracloud.org/api/0/projects'
It returns the list of all projects.
My point is to make one request for all packages instead of doing 10 requests one by one.

Reply:
Ah, okay. I see. Thanks for clarification.